updater/lib/dependabot/updater/error_handler.rb

```diff
@@ -0,0 +1,246 @@
+# frozen_string_literal: true
+
+# This class is responsible for determining how to present a Dependabot::Error
+# to the Service and Logger.
+#
+# TODO: Iterate further on leaner error handling
+#
+# This class is a coarse abstraction of some shared logic that has several flags
+# against it from Rubocop we aren't addressing right now.
+#
+# It feels like this concern could be slimmed down if each Dependabot::Error
+# class implemented a "presenter" method to generate it's own `error-type` and
+# `error-detail` since this never draws attributes from the Updater context.
+#
+# For now, let's just extract it and set it aside as a tangent from the critical
+# path.
+module Dependabot
+  class Updater
+    class ErrorHandler
```

Comment:
I'm appreciating this as a place to put error handling. The reasons for it to change will be limited to new errors and new requirements to handle those errors. That feels right to me, especially compared to the previous location 🙂